admin-node-setup.sh

```diff
@@ -49,15 +49,61 @@ cp $tmp_dir/*.yaml $kube_dir/
 
 rm -rf $tmp_dir
 
-# First time setup of user-configuration for salt-master
+# Create CaaSP config dir
 if [ ! -d "/etc/caasp" ]; then
     mkdir /etc/caasp
 fi
 
+# First time setup of user-configuration for salt-master
 if [ ! -f "/etc/caasp/salt-master-custom.conf" ]; then
     echo "# Custom Configurations for Salt-Master" > /etc/caasp/salt-master-custom.conf
 fi
 
+# Migrate haproxy config path post path change
+if [[ ! -f "/etc/caasp/haproxy/haproxy.cfg" && -f "/etc/haproxy/haproxy.cfg" ]]; then
+    if [ ! -d "/etc/caasp/haproxy" ]; then
+        mkdir /etc/caasp/haproxy
+    fi
+
+    mv /etc/haproxy/haproxy.cfg /etc/caasp/haproxy/haproxy.cfg
+
+    # Add the Velum and Velum-API services to HAproxy
+    cat << EOF >> /etc/caasp/haproxy/haproxy.cfg
+
+listen velum
+        bind 0.0.0.0:80
+        bind 0.0.0.0:443 ssl crt /etc/pki/velum.pem ca-file /etc/pki/ca.crt
+        mode http
+        acl path_autoyast path_reg ^/autoyast$
+        option forwardfor
+        http-request set-header X-Forwarded-Proto https
+        redirect scheme https code 302 if !{ ssl_fc } !path_autoyast
+        default-server inter 10s fall 3
+        balance roundrobin
+        server velum unix@/var/run/puma/dashboard.sock
+
+listen velum-api
+        bind 127.0.0.1:443 ssl crt /etc/pki/velum.pem ca-file /etc/pki/ca.crt
+        mode http
+        option forwardfor
+        http-request set-header X-Forwarded-Proto https
+        default-server inter 10s fall 3
+        balance roundrobin
+        server velum unix@/var/run/puma/api.sock
+EOF
```

Comment:
Why is this append required? If the package installed the new configuration, `/etc/caasp/haproxy/haproxy.cfg` should include https://github.com/kubic-project/caasp-container-manifests/blob/master/config/haproxy/haproxy.cfg with `velum` and `velum-api` already, no? In 2.0 this file does not exist, but during the package upgrade, `/etc/caasp/haproxy/haproxy.cfg` should have been created, no?

Reply:
The packaged file only shows up in a fresh install, during an upgrade, tx update has no access to /etc (well, it does, just those changes will disappear into the ether once the tx-update has exited)